### DISCUSSION TITLE
How to get the current dimensions of a layout

### DISCUSSION THREAD

**Discussion by sortedcord** [Textualize/rich#2705] [Q&A]
Hi, I wanted to know if there was any way to feth the dimensions of a layout that is being displayed on the screen if the size property was not specified when creating the layout?

```py
import rich
from rich.layout import Layout


layout = Layout()
layout.split_column(
    Layout(name="col1"),
    Layout(name="col2"),
)
print(layout['col1'].size)
```

This returns `None`. I wanted to know if I can get the size of the layout that it currently is in which is usually displayed by the placeholder renderable's title.

**Comment by sortedcord** [comment]
I was able to track it down to this method. It seems that it only runs when we are printing something. I haven't seen the code where it is printing using the rich console but if there is any way to access this variable from other scopes then please do let me know

![image](https://user-images.githubusercontent.com/37407370/208380348-dd6129b7-4848-44df-afb0-e08a81924d82.png)

**Reply by nanw1103** [reply]
I think the size is only known during rendering. So you don't get it after layout creation.
